mlflow/models/evaluation/default_evaluator.py

```diff
@@ -708,7 +809,7 @@ def plot_confusion_matrix():
                 "confusion_matrix",
             )
 
-        self._evaluate_custom_metrics()
+        self._log_custom_metrics()
         self._log_metrics()
         self._log_model_explainability()
         return EvaluationResult(self.metrics, self.artifacts)
```

Comment:
This logic is present in both `_evaluate_regressor` and `_evaluate_classifier`. Can we factor it out?

Reply:
I can't think of a fitting function name. Especially after renaming `_log_custom_metrics` to `_evaluate_custom_metrics_and_log_produced_artifacts` as [suggested by Jin](https://github.com/mlflow/mlflow/pull/5405#discussion_r819277286). Does `_prepare_and_generate_evaluation_result` sound good?

Reply:
`_log_and_return_evaluation_result`